app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,15 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = Blog.published.or(Blog.where('user_id = ?', current_user)).find(params[:id])
```

Comment:
文字列でないと指定できない条件以外は、ハッシュ形式で書いたほうが見やすいと思います。

```suggestion
    @blog = Blog.published.or(Blog.where(user: current_user)).find(params[:id])
```